File: aurafit-frontend/src/Models/WishlistItem.ts
import { ProductVariant } from "./ProductVariant";
import {User} from "./User.ts";
import {Product} from "./Product.ts";


export class WishlistItem {
    id: number;
    quantity: number;
    product: Product;
    user: User;

    constructor(id: number, quantity: number, product: Product, user: User) {
        this.id = id;
        this.quantity = quantity;
        this.product = product;
        this.user = user;
    }

}

File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/wishlist/WishlistItem.java
package app.aurafitbackend.wishlist;

import app.aurafitbackend.user.User;
import app.aurafitbackend.product.Product;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "wishlist_items")
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class WishlistItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Long id;

    private Integer quantity;

    @ManyToOne
    private Product product;

//    @ManyToOne
//    private ProductVariant productVariant;

    @ManyToOne
    private User user;

}
